plutus-conformance/src/PlutusConformance/Common.hs

```diff
@@ -146,25 +170,21 @@ expectedToProg txt
         Left _  -> Left txt
         Right p -> Right $ void p
 
--- | Get the tested value. The tested value is either the shown parse or evaluation error,
+-- | Get the tested value from a file (in this case a textual UPLC source
+-- file). The tested value is either the shown parse error or evaluation error,
 -- or a `UplcProg`.
 getTestedValue ::
     UplcEvaluatorFun res
     -> FilePath
     -> IO (Either T.Text res)
-getTestedValue eval dir = do
-    inputFile <- findByExtension [".uplc"] dir
-    case inputFile of
-        [] -> error $ "Input file missing in " <> dir
-        _:_:_ -> error $ "More than 1 input files in " <> dir
-        [file] -> do
```

Comment:
At this point it checks that  there's exactly one `.uplc` file in the directory, but it doesn't check that the filename matches the name of the directory.